axelrod/strategies/finite_state_machines.py

```diff
@@ -57,6 +64,7 @@ def __init__(self, transitions=None, initial_state=None,
             initial_action = C
         super().__init__()
         self.initial_state = initial_state
+        self.state = initial_state
```

Comment:
Needed for the reset test. The cloned player did not have a `self.state` attribute if it had not played yet.